### ISSUE TITLE
Could not load file or assembly 'System.Collections.Immutable... on OS X with latest CLI

### ISSUE THREAD

**Issue by miguellira** [aspnet/cli-samples#31]
Running the latest HelloMvc app against the latest CLI (`1.0.0-beta-002188`) on OS X generates a `System.IO.FileLoadException`.

**NOTE:** This issue does **_not_** affect Windows 10 x64 with the same CLI.

### Steps to reproduce
```
git clone https://github.com/aspnet/cli-samples.git
dotnet -v restore
dotnet -v build
dotnet -v run
```

### Expected behavior

Running website with a static image and the text "Hello World".

### Actual behavior

```console
An unhandled exception has occurred while executing the request
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.IO.FileLoadException: Could not load file or assembly 'System.Collections.Immutable, Version=1.1.37.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. An internal error occurred.
 (Exception from HRESULT: 0x8007054F)
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean noCheck, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& bNeedSecurityCheck)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
   at System.Activator.CreateInstance[T]()
   at Microsoft.Extensions.Options.OptionsCache`1.CreateOptions()
   at System.Threading.LazyInitializer.EnsureInitializedCore[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at Microsoft.AspNetCore.Mvc.Razor.Internal.DefaultRazorViewEngineFileProviderAccessor..ctor(IOptions`1 optionsAccessor)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ScopedCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ScopedCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.TransientCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.TransientCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ScopedCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.AspNetCore.Mvc.Razor.Internal.MvcRazorMvcViewOptionsSetup.ConfigureMvc(IServiceProvider serviceProvider, MvcViewOptions options)
   at Microsoft.Extensions.Options.OptionsCache`1.CreateOptions()
   at System.Threading.LazyInitializer.EnsureInitializedCore[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at Microsoft.AspNetCore.Mvc.ViewEngines.CompositeViewEngine..ctor(IOptions`1 optionsAccessor)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ScopedCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ScopedCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.AspNetCore.Mvc.ViewResult.<ExecuteResultAsync>d__26.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Mvc.Internal.FilterActionInvoker.<InvokeResultAsync>d__44.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Mvc.Internal.FilterActionInvoker.<InvokeResultFilterAsync>d__43.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Internal.FilterActionInvoker.<InvokeAllResultFiltersAsync>d__42.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Mvc.Internal.FilterActionInvoker.<InvokeResourceFilterAsync>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Internal.FilterActionInvoker.<InvokeAsync>d__32.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Mvc.Internal.MvcRouteHandler.<InvokeActionAsync>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.<Invoke>d__7.MoveNext()
```
### Environment data

dotnet --info output:
```console
.NET Command Line Tools (1.0.0-beta-002188)

Product Information:
 Version:     1.0.0-beta-002188
 Commit Sha:  29b567db3f

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.11
 OS Platform: Darwin
 RID:         osx.10.11-x64
```

**Comment by davidfowl** [comment]
I'm not reproducing this on OSX. 

```
Davids-MacBook-Pro:HelloMvc dfowler$ dotnet --info
.NET Command Line Tools (1.0.0-beta-002190)

Product Information:
 Version:     1.0.0-beta-002190
 Commit Sha:  be1c55f061

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.11
 OS Platform: Darwin
 RID:         osx.10.11-x64
```

Lets compare lock files.

**Comment by miguellira** [comment]
@davidfowl I'll try `2190` before sharing out the lock file.

**Comment by miguellira** [comment]
@davidfowl Upgrade to `2190` and still getting the error. Here's my lock file: https://gist.github.com/miguellira/c91c40f671ad7e95e7a4297663ddd8b1

**Comment by miguellira** [comment]
@davidfowl This gets interestring...

Decided to clear out my local nuget cache (`~/.nuget/pacakges`) and retry and now I get this error:
```console
      An unhandled exception has occurred while executing the request
System.TypeInitializationException: The type initializer for 'Microsoft.AspNetCore.Mvc.ModelBinding.BindingSource' threw an exception. ---> System.IO.FileLoadException: Could not load file or assembly 'System.Resources.ResourceManager, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. An internal error occurred.
 (Exception from HRESULT: 0x8007054F)
   at Microsoft.AspNetCore.Mvc.Abstractions.Resources.GetString(String name, String[] formatterNames)
   at Microsoft.AspNetCore.Mvc.ModelBinding.BindingSource..cctor()
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Mvc.ModelBinding.RouteValueProviderFactory.CreateValueProviderAsync(ValueProviderFactoryContext context)
   at Microsoft.AspNetCore.Mvc.Internal.FilterActionInvoker.<InvokeResourceFilterAsync>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Internal.FilterActionInvoker.<InvokeAsync>d__32.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Mvc.Internal.MvcRouteHandler.<InvokeActionAsync>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.<Invoke>d__7.MoveNext()
```
Furthermore, I then killed the process `Ctrl-C` and tried again and got this:
```console
Miguels-MacBook-Pro:HelloMvc miguellira$ dotnet -v run
Project HelloMvc (.NETCoreApp,Version=v1.0) was previously compiled. Skipping compilation.
Running /usr/local/share/dotnet/dotnet exec /Users/miguellira/Documents/Github/aspnet/cli-samples/HelloMvc/bin/Debug/netcoreapp1.0/HelloMvc.dll --additionalprobingpath /Users/miguellira/.nuget/packages
Process ID: 2918
dbug: Microsoft.AspNetCore.Hosting.Internal.WebHost[3]
      Hosting starting
Unhandled Exception: System.AggregateException: One or more errors occurred. (Error -48 EADDRINUSE address already in use) ---> Microsoft.AspNetCore.Server.Kestrel.Networking.UvException: Error -48 EADDRINUSE address already in use
   at Microsoft.AspNetCore.Server.Kestrel.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Networking.UvStreamHandle.Listen(Int32 backlog, Action`4 callback, Object state)
   at Microsoft.AspNetCore.Server.Kestrel.Http.TcpListenerPrimary.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Http.Listener.<>c.<StartAsync>b__6_0(Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.Http.ListenerPrimary.<StartAsync>d__9.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.KestrelEngine.CreateServer(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.KestrelServer.Start[TContext](IHttpApplication`1 application)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.Start()
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.Run(IWebHost host, CancellationToken token, String shutdownMessage)
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.Run(IWebHost host)
   at HelloMvc.Program.Main(String[] args)
```
Same terminal window - not a different one. Guess it takes awhile for Kestrel/Libuv to clean up but it's weird that I never ran into this before.

**Comment by davidfowl** [comment]
This isnt making any sense... Can you clean the packages folder again, clean the output folder, and run?

**Comment by miguellira** [comment]
@davidfowl Not sure what you mean by "output folder", I just cleared my nuget package cache, bin, obj folders and lock file and tried it again. Same error.

**Comment by davidfowl** [comment]
Which error are you getting? bin and obj are the output folders I was referring to.

**Comment by miguellira** [comment]
@davidfowl The binding error: 
```console
An unhandled exception has occurred while executing the request
System.TypeInitializationException: The type initializer for 'Microsoft.AspNetCore.Mvc.ModelBinding.BindingSource' threw an exception. ---> System.IO.FileLoadException: Could not load file or assembly 'System.Resources.ResourceManager, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. An internal error occurred.
 (Exception from HRESULT: 0x8007054F)
```
The `System.Collections.Immutable` seems to have gone away since upgrading to `2190`

**Comment by davidfowl** [comment]
Might be something wrong with your installation. I don't see anything compiled against System.Resources.ResourceManager 4.0.0.0 and it still works for me. Looks like we need one more person to try it :smile:

**Comment by miguellira** [comment]
@davidfowl I've had to uninstall my dotnet cli installation via `pkgutil` in the past, might be worth doing that again and seeing if it clears up.

**Comment by miguellira** [comment]
@davidfowl So I ran `pkgutil --pkgs` and noticed the following:
```console
com.microsoft.dotnet.dev.1.0.0-beta-002060.component.osx.x64
com.microsoft.dotnet.dev.1.0.0-beta-002111.component.osx.x64
com.microsoft.dotnet.dev.1.0.0-beta-002115.component.osx.x64
com.microsoft.dotnet.dev.1.0.0-beta-002171.component.osx.x64
com.microsoft.dotnet.dev.1.0.0-beta-002175.component.osx.x64
com.microsoft.dotnet.dev.1.0.0-beta-002188.component.osx.x64
com.microsoft.dotnet.dev.1.0.0-beta-002190.component.osx.x64
com.microsoft.dotnet.sharedframework.Microsoft.NETCore.App.1.0.0-rc2-23925.component.osx.x64
com.microsoft.dotnet.sharedframework.Microsoft.NETCore.App.1.0.0-rc2-23928.component.osx.x64
com.microsoft.dotnet.sharedframework.Microsoft.NETCore.App.1.0.0-rc2-23929.component.osx.x64
com.microsoft.dotnet.sharedframework.Microsoft.NETCore.App.1.0.0-rc2-23930.component.osx.x64
com.microsoft.dotnet.sharedframework.Microsoft.NETCore.App.1.0.0-rc2-23931.component.osx.x64
com.microsoft.dotnet.sharedhost.1.0.0.component.osx.x64
```
I've run through quite a few CLI installations lately... 

Anyway, I deleted my dotnet cli root directory (`/usr/local/share/dotnet/`), ran `pkgutil --forget` on each of those packages, installed the latest CLI (`1.0.0-beta-002192` as of this post), cleaned out the nuget cache and output folders and I'm still getting the same error.

Since the error is different from my original post, should I close this one and create a new one?

**Comment by miguellira** [comment]
@davidfowl I figured out how to run the sample HelloMvc app without errors. However, I'm not sure why you're not seeing the exception.

Basically when I run `dotnet -v run` the CLI actually executes the following statement:
```console
/usr/local/share/dotnet/dotnet exec /Users/miguellira/Documents/Github/aspnet/cli-samples/HelloMvc/bin/Debug/netcoreapp1.0/HelloMvc.dll --additionalprobingpath /Users/miguellira/.nuget/packages
```
and the `System.TypeInitializationException` error is produced while accessing the site. However, if I omit the `--additionalprobingpath /Users/miguellira/.nuget/packages` flag and simply execute `HelloMvc.dll` it works fine.

The thing is I've nuked my `~/.nuget/packages` directory countless times so it should be probing the latest bits, is it possible that somehow yours works because you haven't?
